### ISSUE TITLE
Allow VM.setMetadata to merge Metadata

### ISSUE THREAD

**Issue by pickypg** [googleapis/nodejs-compute#168]
TL;DR: Provide an optional parameter for `VM.setMetadata` so that it can optionally merge existing metadata for the caller.

Because the `/setMetadata` call requires you to provide both a fingerprint and the entire payload of metadata with the call, `VM.setMetadata` internally does a `VM.getMetadata` call for the fingerprint, but simply ignores the existing metadata (if any) in favor of the incoming metadata.

I would expect a fair number of use cases to be appending metadata based on detected details and since the internal call already fetches the existing metadata it would make sense to provide the option to merge the two sets of metadata together. Otherwise, I am forced to do the `VM.getMetadata` call myself, then merge the metadata, then pass to `VM.setMetadata` only to have it do the exact same `VM.getMetadata` call (note line 767 in particular):

https://github.com/googleapis/nodejs-compute/blob/49a6dcca7cd78b61db6b7be1733aef4f68157500/src/vm.js#L754-L788

I would be happy to submit a PR if there is interest.

#### Environment details

  - OS: macOS 10.13.6
  - Node.js version: 6.14.0 (GCF)
  - npm version: 5.6.0
  - `@google-cloud/compute` version: 0.10

**Comment by stephenplusplus** [comment]
That makes sense! I think we could just default to the patch semantics, as that's what our `setMetadata` methods normally do. A PR would be great!
